pytorch_lightning/loops/epoch/prediction_epoch_loop.py

```diff
@@ -119,12 +118,10 @@ def _predict_step(self, batch: Any, batch_idx: int, dataloader_idx: int) -> None
         """
         # configure step_kwargs
         step_kwargs = self._build_kwargs(batch, batch_idx, dataloader_idx)
-
-        # extract batch_indices and store them
-        self._store_batch_indices(dataloader_idx)
-
         model_ref = self.trainer.lightning_module
 
+        self.current_batch_indices = self._seen_batch_indices[batch_idx] if self._seen_batch_indices else []
```

Comment:
Won't it be better to pop the `current_batch_indices` ?

Reply:
I think we need to preserve the full list, as this gets returned `on_run_end` and then returned from `trainer.predict()`. If we don't need the full list, we could pop I guess.